app/pages/register.vue

```diff
@@ -204,7 +213,13 @@ async function register(event: FormSubmitEvent<Schema>) {
         <USeparator label="ATAU" />
         <p class="text-center">
           Sudah memiliki akun?
-          <ULink to="/login" class="text-primary-600 hover:underline"
+          <ULink
+            :to="
+              redirectPath
+                ? `/login?redirect=${encodeURIComponent(redirectPath)}`
+                : '/login'
+            "
```

Comment:
I found this approach is buggy while maintaining existing query like if you in the `/reading-habits?date=05%2F24%2F2025` page. I encourage you to read the documentation, which we can use query object instead: https://ui.nuxt.com/components/link
![image](https://github.com/user-attachments/assets/0c52d2d4-c496-4ea4-a112-cd822f14b78c)